kernel/page_allocator/src/lib.rs

```diff
@@ -9,8 +9,11 @@
 //! The core allocation function is [`allocate_pages_deferred()`](fn.allocate_pages_deferred.html), 
 //! but there are several convenience functions that offer simpler interfaces for general usage. 
 //!
-//! # Notes and Missing Features
-//! This allocator currently does **not** merge freed chunks (de-fragmentation) upon deallocation. 
+//! # Notes
+//! This allocator only makes one attempt to merge freed chunks (for de-fragmentation)
+//! upon deallocation. It does not iteratively merge adjacent chunks in order to
+//! Free chunks are lazily merged more aggressively upon running out of address space
```

Comment:
the code looks good to me, only here the end of sentence is missing

Reply:
Oh thanks, that's what I get for trying to code while gaming 😅